### DISCUSSION TITLE
Cannot add new property

### DISCUSSION THREAD

**Discussion by AndresdoSantos** [nativewind/nativewind#615] [Q&A]
Hi guys, I'm trying to create a hook to change the application's theme, but I really can't because of the `cannot add a new property` error.

I'm using version `"nativewind": "2.0.11"`

This is the code

```ts
// use-theme.ts

import { useColorScheme } from 'nativewind'

const changeTheme = useCallback(
  (theme?: ColorSchemeSystem) => {
    setColorScheme(theme || colorScheme === 'dark' ? 'light' : 'dark')

    setStorageTheme(theme || colorScheme === 'dark' ? 'light' : 'dark')
  },
  [colorScheme, setColorScheme, setStorageTheme],
)
```

...and in `App.tsx` I call the function to set the theme that is saved within MMKV.

```ts
// App.tsx

useEffect(() => {
  setTheme(theme as ColorSchemeSystem)
}, [setTheme, theme])
```

The problem is that when this function (changeTheme) is called the following error happens: **cannot add a new property**

```
// runtime.js

  }
    if (classNameStyles.childClassNames) {
        childStyles.push(...classNameStyles.childClassNames);
    }
}

```

And I need to update the application to work properly, does anyone know why this happens?

**Comment by shanoysinc** [comment]
did you find a solution?

**Comment by kkx64** [comment]
Happens to me when rotating the device (and assuming because of the breakpoint logic?)

**Comment by danstepanov** [comment]
Closing this discussion, if this is still relevant, feel free to open a new one.
